quiche/src/recovery/congestion/bbr/per_ack.rs

```diff
@@ -258,13 +258,28 @@ fn bbr_check_drain(r: &mut Congestion, bytes_in_flight: usize, now: Instant) {
     }
 
     if r.bbr_state.state == BBRStateMachine::Drain &&
-        bytes_in_flight <= bbr_inflight(r, 1.0)
+        bbr_bytes_in_net(r, bytes_in_flight, now) <= bbr_inflight(r, 1.0)
     {
         // we estimate queue is drained
         bbr_enter_probe_bw(r, now);
     }
 }
 
+fn bbr_bytes_in_net(r: &Congestion, in_flight: usize, now: Instant) -> usize {
+    let edt = r.pacer.next_time().max(now);
+    let interval = edt.saturating_duration_since(now);
+    let interval_delivered = interval.as_secs_f64() * r.bbr_state.btlbw as f64;
+
+    let mut in_flight_at_edt = in_flight;
+
+    // pacing_gain > 1.0
+    if (r.bbr_state.pacing_gain - 1.0).abs() > f64::EPSILON {
+        in_flight_at_edt += r.send_quantum;
+    }
```

Comment:
- Are we missing these [lines](https://elixir.bootlin.com/linux/v6.13.7/source/net/ipv4/tcp_bbr.c#L451-L452) and do we need them?  `if (interval_delivered >= inflight_at_edt) return 0;`

If I am understanding correctly,  `r.send_quantum` is the approximation of what we approximate is buffered. Thats more than I would have expected.

Reply:
The `saturating_sub` in the next line is equivalent to the `if (interval_delivered >= inflight_at_edt) return 0;` expression.

I think that the `send_quantum` computation in the previous lines corresponds to https://elixir.bootlin.com/linux/v6.13.7/source/net/ipv4/tcp_bbr.c#L449-L450